flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapValueState.java

```diff
@@ -30,7 +30,7 @@
  * @param <V> The type of the value.
  */
 public class HeapValueState<K, N, V>
-		extends AbstractHeapState<K, N, V, ValueState<V>>
+		extends AbstractHeapState<K, N, V>
```

Comment:
You could consider to also use the new `getInternal()` and `updateInternal()` methods insider the methods of this class as well to replace direct calls to the `stateTable`.

Reply:
The same also holds for `HeapMapState` etc.